sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1595 @@
+cycle `N` is chosen as the "Last PoX-1" reward cycle, and then the
+"First PoX-2" reward cycle is the subsequent cycle (`N+1`).
+
+This defines three periods of PoX operation:
+
+Period 1 | Period 2 | Period 3
+-- | -- | --
+2.0 Consensus Rules in Effect | 2.1 Consensus Rules enacted, but first PoX-2 reward cycle has not begun | First PoX-2 reward cycle has begun
+_This is the period of time before the 2.1 fork._ | _This is after the 2.1 fork, but before cycle (N+1)._ | _This is after cycle (N+1) has begun. Original PoX contract state will no longer have any impact on reward sets, account lock status, etc._
+
+- Every account that is locked by the original contract for cycle `N`
+  and beyond is unlocked at the end of Cycle `N`.
+- Every account that is locked (whether by PoX-1 or PoX-2) is eligible
+  for a call to `pox-2.stack-extend`, which allows an account to
+  re-lock for some subsequent number of reward cycles while still
+  being locked.
+    - This would also be true for `pox-2.delegate-stack-extend`
+- Calls to PoX 2 which would attempt to create state for a cycle
+  _before_ `(N+1)` will fail
+- Calls to the original PoX contract which would attempt to create
+  state for a cycle `>= N+1` will be made to fail and any state after
+  `N+1` is ignored.  This requires interposing on contract-calls
+  during period 2 and checking the reward cycles arguments: the
+  relevant functions are `stack-stx`, `delegate-stack-stx`,
+  `stack-aggregation-commit`, and `reject-pox`.
+
+### New method: `stack-extend`
+
+This method allows direct stackers to re-lock their funds for up to an additional
+12 cycles *before* their current lock has expired. The target use case for
+this is to allow users to repeatedly stack without a cooldown phase.
+
+This method checks that the stacker is still allowed to stack, that
+the stacker has not delegated to an operator, and that their funds are
+currently stacked. The caller may supply a new reward address for the
+extension.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the stacker's STX account to
+"auto-unlock" at the end of the last extended-to reward cycle.
+
+### New method: `delegate-stack-extend`
+
+This method allows operators to re-lock the funds of one of their
+delegation participants for additional cycles *before* that participants
+lock has expired. The target use case for this is to allow delegation
+operators to repeatedly stack on behalf of their users without cooldowns.
+
+This method checks that the delegator is still authorized on behalf of
+the given stacker (and remains authorized until the end of the
+locked-for period) and that the funds are currently stacked.
+
+Note that, just as with the existing `delegate-stack-stx` function,
+this method locks *but does not commit* the user's STX. The delegation
+operator must invoke `stack-aggregation-commit` to set a reward address
```

Comment:
Is the removal of the minimum for stack-aggregation-commit a breaking change that needs to be listed in this SIP?

Reply:
There is still a minimum Stacking threshold, but once that threshold has been met, the user can increase the amount of STX locked by as little as 1 uSTX with `stack-increase` or `delegate-stack-increase`.

Reply:
My comment was about `min_increment_ustx` that is enforced on `stacks-aggressition-commit`

Reply:
`stacks-aggregation-commit` still uses `can-stack-stx` internally, which verifies that at least `(get-stacking-minimum)` uSTX are stacked by the pool operator for the given `reward-cycle`.

This section is only saying that `delgate-stack-extend` is added so the user can commit their STX to later reward cycles while STX are locked.  The number of STX they have locked does not change.  The pool operator must call `stack-aggregation-commit` to commit the extension on the new reward cycles in which the STX are now committed.

Reply:
I have hoped that pool operators can add more pool members after a first call to `stack-aggregation-commit`. 
I understand your comment that pool members can't increase with 1 uSTX - even though some of their STX are already above the threshold as part of the pool - if the sum of increases by all pool members is below the threshold.

Reply:
> I have hoped that pool operators can add more pool members after a first call to stack-aggregation-commit.

Ah, I see.  Let me see if we can address this in the code.  I think that once the given PoX address has cleared the minimum Stacking threshold, it should be safe to increment the stacked amount by 1 uSTX.

Reply:
Tracking this in an issue: https://github.com/stacks-network/stacks-blockchain/issues/3360